### ISSUE TITLE
H-Purifier Hoover Air Purifier not added

### ISSUE THREAD

**Issue by paolograssi** [gvigroux/hon#22]
Hello!!!
First of all, great development, thanks!!

I would like to report to you that my Hoover H-Purifier 700 is not added automatically.

Is there any help I can give you to add this (I am a developer)?

**Comment by OcioRanocio** [comment]
Hi there,

I have an Hoover H-Purifier 300 and I succeeded in importing some sensors and making the purifier start and stop.
It's still a work in progress and I'm not a programmer, but if it helps I can share my code.

Ciao

**Comment by OcioRanocio** [comment]
Look for my fork here. Please let me know if it works with yours :-)
Sei italiano? Are you italian?
The difficult part for me about sensors, is their values, I would need some kind of conversion tale, I mean, I can take their values but I don't often know what it represents.

**Comment by psychowood** [comment]
> Look for my fork here. Please let me know if it works with yours :-) Sei italiano? Are you italian? The difficult part for me about sensors, is their values, I would need some kind of conversion tale, I mean, I can take their values but I don't often know what it represents.

_Conterraneo_ of yours here :)  testing your [fork](https://github.com/OcioRanocio/hon) with a Hoover H-Purifier 700.

- Device was discovered correctly
- I only have 5 sensors, reporting correct values

<img width="1001" alt="image" src="https://user-images.githubusercontent.com/115389/209435001-ae2a8850-8966-4161-bcfe-3aade9265a18.png">

- Calling services sets correctly the power status and mode, it would be nice to add the relative switches/actions in the device entities 

- The 700 model has some additional options (essence diffuser, humidifier, Co/temperature/humidity sensors, light and sound settings)

I would gladly try to contribute if you can give me some directions in how to start (I'm quite technical but I've got no experience with HA addons).

**Comment by OcioRanocio** [comment]
Ciao! I'm not a developer and this is the first time I work with Home Assistant/Python. Anyway I've just understood how the component works and did a try and error method. You have to check what the server answers when the component start. You can see that in the logs deleting 7 from the appliancetypeid (7 is the Air Purifier)
![Screenshot_20221228-163707_Chrome](https://user-images.githubusercontent.com/52009430/209837623-5f13a4e8-a784-42bf-b72e-a48f43783958.jpg)

**Comment by Dalewn** [comment]
Hi there,

I also own an H-Purifier 700. I found the repo of @artystazst pretty helpful as a starting point, but it doesn't include all actions the 700 offers.
I am missing control for the humidifier and the threshold values.
Also I am relatively versed in programming but have no clue about python yet.

Is there a way to call the api via REST or sth? I did scroll through the auth procedures in the code but it all looked a little odd to me tbh.
